pleskdistup/actions/common_checks.py

```diff
@@ -142,3 +144,109 @@ def __init__(self):
 
     def _do_check(self) -> bool:
         return subprocess.run(["/bin/fuser", "/var/lib/apt/lists/lock"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL).returncode != 0
+
+
+class MinFreeDiskSpaceViolation(typing.NamedTuple):
+    """Information about a filesystem with insufficient free disk space."""
+    dev: str
+    """Device name."""
+    req_bytes: int
+    """Required space on the device (in bytes)."""
+    avail_bytes: int
+    """Available space on the device (in bytes)."""
+    paths: typing.Set[str]
+    """Paths belonging to this device."""
+
+
+class AssertMinFreeDiskSpace(action.CheckAction):
+    """Check if there's enough free disk space.
+
+    Args:
+        requirements: A dictionary mapping paths to minimum free disk
+            space (in bytes) on the devices containing them.
+        name: Name of the check.
+        description: Description to show if the assertion is violated. If ends
+            with a colon, list of violations will be added automatically.
+    """
+    violations: typing.List[MinFreeDiskSpaceViolation]
+    """List of filesystems with insiffucient free disk space."""
+    _description: str
+
+    def __init__(
+        self,
+        requirements: typing.Dict[str, int],
+        name: str = "check if there's enough free disk space",
+        description: str = "There's not enough free disk space:",
+    ):
+        self.requirements = requirements
+        self.name = name
+        self._description = description
+        self.violations = []
+
+    @property
+    def description(self) -> str:
+        """Description of violations or empty string.
+
+        Can be set to change the message template.
+        """
+        if not self.violations:
```

Comment:
It seems like this situation should never occur. If it does, I would prefer to raise an exception here or, at the very least, return a clear error message such as: "Something has gone very wrong. Please contact us with the feedback archive." Otherwise, there is a chance that the customer may receive information about not meeting conditions without an explanation of the conditions themselves.

Reply:
This situation would occur normally if the check haven't been performed yet or passed.

Reply:
I think this shouldn't happen. Descriptions are supposed to explain the problem that was found and provide clear instructions on how to fix it. It seems odd to receive a description of a problem that hasn't been detected yet.

Reply:
Currently unlikely, but it's a pain to work with classes designed for one particular use case in mind, omitting obvious alternatives.
Also, firing an exception in such case won't make user's situation better, because they won't get the description anyways.

Reply:
> omitting obvious alternatives.
Could you please provide a obvious example of when it's necessary to set a description outside of a check action?

> pain to work with classes designed for one particular use case
There is always a tradeoff between 'pain to work with' and 'pain to maintain'. I believe a class should do exactly the things it was destined for. Otherwise we could add encryption into checker because "it could be useful, why not".

> Also, firing an exception in such case won't make user's situation better, because they won't get the description anyways.
IIRC we will show an error message in this case. Anyway we should have clear description what should be done by customer to solve the problem.

Additionally, I'm not sure other check classes are ready for such way of use

Reply:
I'm not talking about setting the description outside of a check action, I'm just saying that the description can be legitimately empty, if the check hasn't been performed yet or passed. Please not that the property code has been dropped in further commits in favor of direct assignment to the `description` attribute.